### PR TITLE
Shoving someone into a vendor now informs people around them.

### LINKED ISSUE #26072: The Proc For Telling You Who Shoved Who Into an Object Doesn't Work With Vendors
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1633

### Issue Description

When you shove someone into a table or wall, you get told by `/datum/species/proc/disarm` in species.dm. This doesn't function when vendorshoving someone.

### What did you expect to happen?

To be told who did a vendorshove.

### What happened instead?

I am only told that the victim has been crushed.

### Why is this bad/What are the consequences?

In crowded areas, ESPECIALLY during the late arrival station trait, it is impossible to identify who did a vendorshove based on the chat. It also happens a lot during said station trait.

### Steps to reproduce the issue.

Shove someone into vendor.

### When did the problem start happening?

God knows.

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by Spaghetti-bit**
## What Does This PR Do

Fixes #26072

Shoving someone into a vendor now informs people around them who shoved the poor sucker.

## Why It's Good For The Game

Fixes an oversight.

On late arrival station traits, shove spamming is an issue, you can tell who shoves who into a wall but... not into a vendor meaning that the crime is often let go. People scrambling out of the shuttle like that and shoving one another sporadically like maniacs is... not good.

## Images of changes

N/A

## Testing

There was a lot of moth-on-moth violence. I shoved a moth into a vendor! And another! And another!

## Changelog
:cl:
fix: Fixed vendors not announcing who shoved who into them.
/:cl:

### CODE REVIEW

**Review by Fordoxia — APPROVED** [review]
Cool and epic.

Approved for the crunch.

File: code/game/machinery/vendors/vending.dm [hunk 1]
```diff
@@ -980,9 +980,15 @@
 	if(!HAS_TRAIT(attacker, TRAIT_PACIFISM))
 		add_attack_logs(attacker, target, "shoved into a vending machine ([src])")
 		tilt(target, from_combat = TRUE)
+		target.visible_message("<span class='warning'>[attacker] slams [target] into [src]!</span>", \
+								"<span class='userdanger'>You get slammed into [src] by [attacker]!</span>", \
+								"You hear a loud crunch.")
```
Comment (Fordoxia) [inline code/game/machinery/vendors/vending.dm:985]:
```suggestion
		target.visible_message("<span class='danger'>[attacker] slams [target] into [src]!</span>", \
								"<span class='userdanger'>You get slammed into [src] by [attacker]!</span>", \
								"<span class='danger'>You hear a loud crunch.</span>")
```

File: code/game/machinery/vendors/vending.dm [hunk 2]
```diff
@@ -980,9 +980,15 @@
...
 	else if(HAS_TRAIT_FROM(attacker, TRAIT_PACIFISM, GHOST_ROLE))  // should only apply to the ghost bar
 		add_attack_logs(attacker, target, "shoved into a vending machine ([src]), but flattened themselves.")
 		tilt(attacker, crit = TRUE, from_anywhere = TRUE) // get fucked
+		target.visible_message("<span class='warning'>[attacker] tries to slam [target] into [src], but falls face first into [src]!</span>", \
+								"<span class='userdanger'>You get pushed into [src] by [attacker], but narrowly move out of the way as it tips over ontop of [attacker]!</span>", \
+								"You hear a loud crunch.")
```
Comment (Fordoxia) [inline code/game/machinery/vendors/vending.dm:991]:
```suggestion
		target.visible_message("<span class='danger'>[attacker] tries to slam [target] into [src], but falls face first into [src]!</span>", \
								"<span class='userdanger'>You get pushed into [src] by [attacker], but narrowly move out of the way as it tips over ontop of [attacker]!</span>", \
								"<span class='danger'>You hear a loud crunch.</span>")
```

File: code/game/machinery/vendors/vending.dm [hunk 3]
```diff
@@ -980,9 +980,15 @@
...
 		add_attack_logs(attacker, target, "shoved into a vending machine ([src])")
 		tilt(target, from_combat = TRUE)
+		target.visible_message("<span class='warning'>[attacker] slams [target] into [src]!</span>", \
+								"<span class='userdanger'>You get slammed into [src] by [attacker]!</span>", \
+								"You hear a loud crunch.")
 	else if(HAS_TRAIT_FROM(attacker, TRAIT_PACIFISM, GHOST_ROLE))  // should only apply to the ghost bar
 		add_attack_logs(attacker, target, "shoved into a vending machine ([src]), but flattened themselves.")
 		tilt(attacker, crit = TRUE, from_anywhere = TRUE) // get fucked
+		target.visible_message("<span class='warning'>[attacker] tries to slam [target] into [src], but falls face first into [src]!</span>", \
+								"<span class='userdanger'>You get pushed into [src] by [attacker], but narrowly move out of the way as it tips over ontop of [attacker]!</span>", \
```
Comment (lewcc) [inline code/game/machinery/vendors/vending.dm:990]:
```suggestion
								"<span class='userdanger'>You get pushed into [src] by [attacker], but narrowly move out of the way as it tips over on top of [attacker]!</span>", \
```

**Review by Fordoxia — APPROVED** [review]
I suppose having the pacifist smash only be a warning is acceptable.

**Review by Sirryan2002 — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Pull request merged**